packages/core/core/src/public/Symbols.js

```diff
@@ -5,95 +5,148 @@ import type {
   Symbols as ISymbols,
   SourceLocation,
 } from '@parcel/types';
+import type {Asset, Dependency} from '../types';
 
-let parcelSymbolsToSymbols: WeakMap<
-  Map<Symbol, {|local: Symbol, loc: ?SourceLocation|}>,
-  Symbols,
-> = new WeakMap();
+import nullthrows from 'nullthrows';
+
+let valueToSymbols: WeakMap<Asset, Symbols> = new WeakMap();
 
 export class Symbols implements ISymbols {
   /*::
   @@iterator(): Iterator<[Symbol, {|local: Symbol, loc: ?SourceLocation|}]> { return ({}: any); }
   */
-  #symbols; // Map<Symbol, {|local: Symbol, loc: ?SourceLocation|}>
+  #value; // Asset
 
-  constructor(symbols: Map<Symbol, {|local: Symbol, loc: ?SourceLocation|}>) {
-    let existing = parcelSymbolsToSymbols.get(symbols);
+  constructor(asset: Asset) {
+    let existing = valueToSymbols.get(asset);
     if (existing != null) {
       return existing;
     }
 
-    this.#symbols = symbols;
-    parcelSymbolsToSymbols.set(symbols, this);
+    this.#value = asset;
+    valueToSymbols.set(asset, this);
   }
 
   get(exportSymbol: Symbol): ?{|local: Symbol, loc: ?SourceLocation|} {
-    return this.#symbols.get(exportSymbol);
+    return this.#value.symbols?.get(exportSymbol);
   }
 
   hasExportSymbol(exportSymbol: Symbol): boolean {
-    return this.#symbols.has(exportSymbol);
+    return Boolean(this.#value.symbols?.has(exportSymbol));
   }
 
   hasLocalSymbol(local: Symbol): boolean {
-    for (let s of this.#symbols.values()) {
-      if (local === s.local) return true;
+    if (this.#value.symbols) {
+      for (let s of this.#value.symbols.values()) {
+        if (local === s.local) return true;
+      }
     }
     return false;
   }
 
   // $FlowFixMe
   [Symbol.iterator]() {
-    return this.#symbols[Symbol.iterator]();
+    return this.#value.symbols
+      ? this.#value.symbols[Symbol.iterator]()
+      : {
```

Comment:
Maybe turn this into a function called `createEmptyIterator` which returns this object. Would make this clearer, and could be used in the same way this is used below too.